jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/InvalidURIRule.java

```diff
@@ -119,6 +120,9 @@ protected boolean isValidChar(int codepoint)
         if (LOG.isDebugEnabled())
             LOG.debug("{} {} {} {}", Character.charCount(codepoint), codepoint, block, Character.isISOControl(codepoint));
 
+        if (codepoint == REPLACEMENT_CHAR_CODEPOINT)
```

Comment:
If the client sends a replacement char, then what?

Eg:

```
GET /foo/%EF%BF%BD HTTP/1.1
Host: local
Connection: close
```

Reply:
Then the decoded path will include the replacement character, but the boolean _hasReplacement will be false.

Reply:
We don't have access to the Utf8StringBuilder or Utf8Appendable at this point to ask the .hasReplacement() method.
We have the post-decoded URI.

Reply:
Oh, sorry, I did not see that we are in a Rule.
What is this Rule for? Why is replacement character considered invalid?   We should consider a URI invalid if when we parsed it it had bad UTF8 code points, not because it has replacement characters.

Unless a replacement character is considered invalid regardless of where it came from